configs/components/puppet.rb

```diff
@@ -236,4 +237,56 @@
   if platform.is_eos?
     pkg.link "#{settings[:sysconfdir]}", "#{settings[:link_sysconfdir]}"
   end
+
+  pkg.install_file "ext/hiera/hiera.yaml", File.join(settings[:puppet_codedir], 'environments', 'production', 'hiera.yaml')
+  pkg.configfile File.join(settings[:puppet_codedir], 'environments', 'production', 'hiera.yaml')
+  pkg.configfile File.join(configdir, 'hiera.yaml')
+
+  unless platform.is_windows?
+    old_hiera = File.join(settings[:puppet_codedir], 'hiera.yaml')
+    new_hiera = File.join(configdir, 'hiera.yaml')
```

Comment:
@jcoconnor per @geoffnichols's comment on PA-1139, we should not be overwriting `/etc/puppetlabs/puppet/hiera.yaml` if it already exists. So this needs to also backup `$new_hiera` if it exists on pre-install, and then copy it back on post-install, unconditionally